### ISSUE TITLE
rare lemma warnings

### ISSUE THREAD

**Issue by nschneid** [amir-zeldes/gum#122]
The first one is actually a tag error (should be `-LSB-`)

```
! rare lemma [ for (/-LRB- in GUM_whow_languages-49 (majority: ()

! rare lemma cuz for Because/IN in GUM_vlog_lipstick-71 (majority: because)

! rare lemma Chatnam for Chatham/NNP in GUM_voyage_chatham-2 (majority: Chatham)

! rare lemma Hell for Hells/NNP in GUM_voyage_oakland-4 (majority: Hells)

! rare lemma Tate for Tate/NNP in GUM_speech_remarks-10 (majority: Tale)

! rare lemma alreet for alright/UH in GUM_conversation_atoms-3 (majority: alright)

! rare lemma alroot for alright/UH in GUM_conversation_atoms-3 (majority: alright)

! rare lemma despite for although/IN in GUM_academic_theropod-37 (majority: although)

! rare lemma cuz for because/IN in GUM_vlog_lipstick-83, GUM_vlog_lipstick-76 (majority: because)

! rare lemma in for by/IN in GUM_voyage_cuba-23 (majority: by)

! rare lemma on for in/IN in GUM_vlog_radiology-26, GUM_academic_lighting-21, GUM_academic_lighting-7 (majority: in)

! rare lemma uncompleted for incomplete/JJ in GUM_whow_languages-19 (majority: incomplete)

! rare lemma mildly for mildly/RB in GUM_interview_cocktail-57 (majority: mild)

! rare lemma from for on/IN in GUM_whow_languages-22 (majority: on)

! rare lemma positing for positioning/NN in GUM_whow_ballet-32 (majority: positioning)

! rare lemma recommed for recommended/VBN in GUM_whow_languages-8 (majority: recommend)

! rare lemma set for sat/VBD in GUM_conversation_lambada-59, GUM_conversation_lambada-60 (majority: sit)

! rare lemma it for them/PRP in GUM_textbook_cognition-26, GUM_vlog_radiology-28 (majority: they)

! rare lemma on for to/IN in GUM_interview_peres-48 (majority: to)

! rare lemma who for whoever/WP in GUM_news_taxes-21 (majority: whoever)
```

**Comment by nschneid** [comment]
Note that this is from neaten.py which uses the CorrectForm if provided as the wordform. As "Cuz" is corrected to "Because", its lemma should be "because".

**Comment by amir-zeldes** [comment]
A couple of these are genuine errors ("recommed" is indeed not a valid lemma), but in general I don't think I would treat the corrections as the basis for lemmatization. If someone says "when I was on Croatia", that's not standard English, and should be corrected to "in Croatia". However assuming they genuinely said "on" (as in GUM_vlog_radiology-26), then I think the lemma of that word is still "on". 

In other words, there can be incorrect usages which are wrong only on the form level, but the lemma is correct (e.g. "recommed"), but there can also be errors where the wrong lexical item was chosen, in which case its lemma is also going to be 'wrong' (but it's still the correct lemma to associate with that word, in this case "on").

**Comment by nschneid** [comment]
I always understood the lemma and the CorrectForm to correspond to the same word type. The lemma is the canonical representation of the word. If Typo=Yes/CorrectForm are used for word choice errors like the wrong choice of preposition, shouldn't the lemma be what would go in a canonical version of the sentence, i.e. the corrected sentence? Though TBH I'm not sure this kind of correction (as it doesn't affect the morphological features or deps and isn't a misspelling or abbreviation) belongs in UD.

**Comment by nschneid** [comment]
To put it another way: I don't think it's a goal of most UD treebanks to insert missing determiners or the like. Our goal is to use deprels and features to analyze the grammatical structure that is there, and Typo=Yes/CorrectForm adjust when the form is mismatched to those goals. We're not claiming (I don't think) that any tree without Typo markings is a fully grammatical sentence.

**Comment by amir-zeldes** [comment]
This issue is complicated by the fact that GUM's error annotations are not native UD. They use TEI XML's `<sic>` tags (which are not even tokenwise), and the correction methodology goes back to target hypothesis-based approaches in learner corpora. So in GUM, the correct forms definitely do aspire to something like a fully grammatical sentence, though the interpretation of that is non-trivial (for example repairs using normal English words, as well as filled pauses, are all considerd normal).

I don't think I want to delete the "on" for "in" type of correction in GUM (there are many cases, and it's nice to have this), and I also don't think it's right to lemmatize "on" to "in" in such cases. Maybe the right approach is to have multiple different types of errors, since these are clearly not 'typos', and GUM annotates errors equally for spoken and written language. But it seems weird to have distinct CorrectForm annotations to indicate the corrected variant. In a sense, I think GUM is truer than OOB UD to what I'd expect from the term CorrectForm, but what it has in Typo definitely goes beyong typos.

**Comment by nschneid** [comment]
Hmm. What about adding a new Style feature, e.g. `Style=Nonnative`, instead of Typo=Yes? I.e. treating it as a sociolinguistic variable affecting the word form (like dialect/register, for which there are other `Style` values). That way tools could recognize that the CorrectForm is just a fluency correction, not a correction that pertains to the structure or lemma?

**Comment by amir-zeldes** [comment]
The style is definitely not non-native. My impression of the person uttering the Croatia example was that they were native, and native speakers say weird things all the time. Learner corpus research has shown repeatedly that native control data needs to be annotated for error to get a real sense of errors which distinguish L2 speakers vs ones that natives make just as often.

I also think it's a bit odd to tie lemma to CorrectForm, since all TBs have lemmas, but very few have CorrectForm. In other words, tools should expect the lemma to be based on the normal form. In fact it's a bit weird to expect lemmatizers, which generally use form and POS as inputs, to internally reconstruct a correction when they are not trained on CorrectForm. I view CorrectForm and Typo as lower down the stream than error annotation, though of course in cases like "recommed" we are hoping the lemmatizer will figure it out based on the form an context alone.

**Comment by nschneid** [comment]
I see that in principle they could be different—it's just that 99+% of the time, the CorrectForm is there because of a typo, so statistically speaking it will be a less noisy predictor of the lemma than the misspelled wordform.

Also, I interpret the name "CorrectForm" to mean that we are talking about the same word type, it is just the form of the word that needs correcting.

What about something like CorrectWordChoice instead of CorrectForm for the ones where it is actually a different lemma?

Or: CorrectLemma alongside the CorrectForm? That way there are 2 forms and 2 lemmas, and we know which ones correspond.

**Comment by amir-zeldes** [comment]
I've thought about this and it's a nice topic to devote more attention to in the future, but I don't have the resources for a more fine-grained treatment at the moment. I fixed the actual annotation error cases, but in terms of adding annotations like CorrectLemma this will have to be wontfix for the moment...
